compiler/src/iree/compiler/API/Internal/IREEGPUDialectCAPI.cpp

```diff
@@ -213,3 +214,65 @@ MlirAttribute ireeGPULoweringConfigAttrGetAttributes(MlirAttribute attr) {
                   unwrap(attr))
                   .getAttributes());
 }
+
+ireeGPUTileSizes ireeGPULoweringConfigAttrGetTileSizes(MlirAttribute attr) {
+  assert(ireeAttributeIsAGPULoweringConfigAttr(attr));
+  ireeGPUTileSizes tilesizes = {};
+  auto loweringConfigAttr =
+      llvm::cast<mlir::iree_compiler::IREE::GPU::LoweringConfigAttr>(
+          unwrap(attr));
+
+  llvm::SmallVector<int64_t> workgroups =
+      loweringConfigAttr.getWorkgroupTileSizes();
+  tilesizes.workgroupTileSizes = workgroups.data();
+  tilesizes.numWorkgroupTileSizes = workgroups.size();
+
+  llvm::SmallVector<int64_t> reductions =
+      loweringConfigAttr.getStaticTilingLevelSizes(
+          llvm::to_underlying(
+              mlir::iree_compiler::IREE::GPU::TilingLevel::Reduction),
+          nullptr);
+  tilesizes.reductionTileSizes = reductions.data();
+  tilesizes.numReductionTileSizes = reductions.size();
```

Comment:
This returns dangling pointers. We should use the data from the attr storage -- these two should be array attributes.